apps/remote_control/lib/lexical/remote_control/search/indexer/extractors/struct_reference.ex

```diff
@@ -0,0 +1,103 @@
+defmodule Lexical.RemoteControl.Search.Indexer.Extractors.StructReference do
+  alias Lexical.Document
+  alias Lexical.Document.Position
+  alias Lexical.Document.Range
+  alias Lexical.RemoteControl.Analyzer
+  alias Lexical.RemoteControl.Search.Indexer.Entry
+  alias Lexical.RemoteControl.Search.Indexer.Source.Reducer
+  alias Lexical.RemoteControl.Search.Subject
+
+  require Logger
+
+  @struct_fn_names [:struct, :struct!]
+
+  # Handles usages via an alias, e.g. x = %MyStruct{...} or %__MODULE__{...}
+  def extract(
+        {:%, _, [struct_alias, {:%{}, _, _struct_args}]} = reference,
+        %Reducer{} = reducer
+      ) do
+    case expand_alias(struct_alias, reducer) do
+      {:ok, struct_module} ->
+        {:ok, entry(reducer, struct_module, reference)}
+
+      _ ->
+        :ignored
+    end
+  end
+
+  # Call to Kernel.struct with a fully qualified module e.g. Kernel.struct(MyStruct, ...)
+  def extract(
+        {{:., _, [kernel_alias, struct_fn_name]}, _, [struct_alias]} = reference,
+        %Reducer{} = reducer
+      )
+      when struct_fn_name in @struct_fn_names do
+    with {:ok, Kernel} <- expand_alias(kernel_alias, reducer),
+         {:ok, struct_module} <- expand_alias(struct_alias, reducer) do
+      {:ok, entry(reducer, struct_module, reference)}
+    end
+  end
+
+  # handles calls to Kernel.struct e.g. struct(MyModule) or struct(MyModule, foo: 3)
+  def extract({struct_fn_name, _, [struct_alias | _] = args} = reference, %Reducer{} = reducer)
+      when struct_fn_name in @struct_fn_names do
+    reducer_position = Reducer.position(reducer)
+    imports = Analyzer.imports_at(reducer.analysis, reducer_position)
+    arity = length(args)
+
+    with true <- Enum.member?(imports, {Kernel, :struct, arity}),
```

Comment:
I think it should be this to also handle `struct!`?

```suggestion
    with true <- Enum.member?(imports, {Kernel, struct_fn_name, arity}),
```